### PR TITLE
Phase-Difference field map ingress/parsing fixes

### PR DESCRIPTION
**Pull request by sgiavasis**
## Fixes

Related to #1582 by @sgiavasis 

## Description
Further fixes for input parsing of field map files based on test runs with a wider range of input datasets.

## Technical details
Includes both phase-difference (`PhaseDiff`) and phase-direction (`Blip`) distortion correction field map ingress.

## Tests
- [x] Can run both `PhaseDiff` and `Blip` distortion corrections with multiple data releases (`Site-CBIC` and `Site-RU` of [HBN](http://fcon_1000.projects.nitrc.org/indi/cmi_healthy_brain_network/), for example).

## TO-DO
- [x] Still need to resolve an issue with `bytes`-type phase direction coming into [convert_pedir](https://github.com/FCP-INDI/C-PAC/blob/fix/distcor_fixes/CPAC/registration/registration.py#L688-L704).
- [x] Need to update changelog.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the `develop` branch of the repository. 
- [x] My commit messages follow best practices.
- [x] My code follows the established code style of the repository.
- [ ] I added tests for the changes I made (if applicable).
- [x] I updated the changelog.
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: CPAC/utils/datasource.py [hunk 1]
```diff
@@ -409,8 +409,8 @@ def ingress_func_metadata(wf, cfg, rpool, sub_dict, subject_id,
 
             fmap_rp_list.append(key)
 
-            if key == "diff_phase" or key == "diff_mag_one" or \
-                            key == "diff_mag_two":
```
Comment (shnizzedy) [inline CPAC/utils/datasource.py:413]: `"diff_mag_two"` is not relevant here anymore?

**Review by shnizzedy — APPROVED** [review]
LGTM. Are you thinking merge this and do the TODOs separately?

File: CPAC/utils/datasource.py (hunk 1 above)
Comment (sgiavasis) [reply CPAC/utils/datasource.py:413]: @amygutierrez resolved the `bytes` TO-DO (I just checked it off).
I'll update the changelog shortly - thanks for reminder!

**Comment by sgiavasis** [comment]
> LGTM. Are you thinking merge this and do the TODOs separately?

@amygutierrez resolved the `bytes` TO-DO (I just checked it off).
I'll update the changelog shortly - thanks for reminder!

File: CPAC/utils/datasource.py (hunk 1 above)
Comment (sgiavasis) [reply CPAC/utils/datasource.py:413]: Not with the way we implement this type of distcor. But we want to revisit, especially regarding when we apply the unwarp (probably should apply before co-reg).

Comment (sgiavasis) [reply CPAC/utils/datasource.py:413]: Second diff mag can be used in some cases.

**Pull request merged**